src/Native/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -506,6 +509,10 @@ extern "C" void PalAttachThread(void* thread)
 extern "C" bool PalDetachThread(void* thread)
 {
     UNREFERENCED_PARAMETER(thread);
+    if (g_threadDetachCallback != nullptr)
```

Comment:
> In this case it would be hard to guarantee that the callback happens before PalDetachThread. 

That's one of the reasons why the callback should not be calling back into managed code. The work done in the callback should be very minimal - kind of like what you would do in signal handler: set a bit in unmanaged memory, write something to a pipe, etc.

> Though I guess we might reinstall the destroyed value back into TLS like CoreCLR does

This is fragile hack. I do not think we should be replicating it here.